astropy/cosmology/core.py

```diff
@@ -21,6 +21,8 @@
 from .parameter import Parameter
 
 if TYPE_CHECKING:  # pragma: no cover
+    from collections.abc import Mapping
```

Comment:
I'm confused, why is this not imported unconditionally? Why the need for `TYPE_CHECKING`?

Reply:
If an import is used only in a type-checking context, without a runtime requirement, then we don't need to import it during runtime. I advocated to get this check into Ruff because it helps to minimize runtime imports and import cycles.

Reply:
For example, consider type checking a function that can take a SciPy object, but we don't have SciPy as a runtime dependency. Separating runtime from non-runtime imports would allow us to both type hint the function and not have the runtime dependency. Ruff now sorts our imports and (can or does, I forget which setting) remove unused imports. It also tells us which imports are used in what context. Hopefully autofix is coming soon, so type-checking-only imports will be auto-sorted into `if TYPE_CHECKING` blocks.

Reply:
OK, thanks, that makes sense. I was surprised partially since now the annotation refers to an object that is not necessarily imported. But perhaps this is where the lazy evaluation of annotations that I vaguely recall being discussed comes in?

Anyway, if it works, that's great!

Reply:
Yes, the lazy evaluation of annotations is a complicated landscape. 
https://peps.python.org/pep-0563/ vs https://peps.python.org/pep-0649/. When it's eventually settles, that'll be great.